vignettes/rhino-fluent.Rmd

```diff
@@ -0,0 +1,556 @@
+Let us style the data table area.
+
+The following code does 2 things -
+
+-   Creates an elevated card around the data table area
+
+-   Confines the height of the table area
+
+```         
+div(
+      class = "ms-depth-64", 
+      style = "height: 500px; overflow: auto;",
+      uiOutput(ns("datatable"))
+    )
+```
+
+The above code creates a **`div`** element with the following attributes -
+
+**`class = "ms-depth-64"`**assigns the CSS class **`ms-depth-64`** to the **`div`** element, which may define the element's visual appearance and behavior.
+
+**`style = "height: 500px; overflow: auto;"`** sets the inline CSS style for the **`div`** element.
+Specifically, it sets the height of the **`div`** element to be 500 pixels and enables scrolling within the **`div`** element when its contents overflow its boundaries.
+
+So, this is how the UI should look now -
+
+```{r, echo=FALSE}
+knitr::include_graphics("images/rhino_fluent_gif_2.gif")
+```
+
+## 5. Render a Bar Chart with logic in a helper function
+
+We will now add a bar chart that plots the deal amounts closed by the sales reps. For this, we will code a helper function and keep it inside the `app/logic` folder.
+
+### Helper function in the logic folder of the `rhino` project
+
+-   Go to the `app/logic` folder and create a new `.R` file.
+    Lets name it `barchart.R`
+
+-   Install `gglplot2` and `plotly` packages for this purpose.
+    Please follow the same steps that were followed previously during and after new package installation.
+
+-   Here is the code -
+
+    ```         
+    box::use(
+      ggplot2[ggplot, aes, geom_bar, labs, theme_light],
+    )
+
+    #' @export
+    generate_barchart <- function(filtered_data) {
+            ggplot(filtered_data, aes(x = rep_name, y = deal_amount, fill = rep_name)) +
+            geom_bar(stat = "identity") +
+            labs(x = "Sales Rep Name", y = "Deal Amount") +
+            theme_light()
+    }
+    ```
+
+-   The above code block loads the **`ggplot2`** package and imports several functions from it.
+    It then defines a function called **`generate_barchart()`** that takes a **`filtered_data`** argument as input.
+    The function uses the **`ggplot()`** function to create a bar chart with **`x = rep_name`** and **`y = deal_amount`**, where **`rep_name`** and **`deal_amount`** are columns in the **`filtered_data`** data frame.
+    The **`fill`** aesthetic is also set to **`rep_name`**, which maps each bar to a unique color based on the **`rep_name`** variable.
+    The function also adds labels to the x and y axes and applies the **`theme_light()`** theme to the plot.
+    Finally, the function is exported with **`#' @export`**, which makes it available to other packages and scripts that import this one.
+
+### Call the helper function in the desired `rhino` module
+
+-   Now let us go back to the `app\view\datatable.R` file.
```

Comment:
Why the chart is not in a separate module?

Reply:
The logic has been separated as a standalone function inside logic folder. Did not create a separate module for this in view folder to -
- reduce complexity
- include an inclusive example to call a logic function in a view module that gets called in main file